New msprime calling convention

hyanwong opened the pull request.

jeromekelleher commented:
I don't think we should change the code until msprime 1 is released (or at least we have a alpha/beta). Currently this isn't being run as part of CI, so it's clearly not being tested.

hyanwong commented:
ISTR that I changed it because I was getting warnings when running `nosetests` when I had installed a GitHub version of msprime. But let's leave this open until msprime 1?

jeromekelleher commented:
> ISTR that I changed it because I was getting warnings when running nosetests when I had installed a GitHub version of msprime.

Yes, but we can't change the code for everyone until msprime is released. Probably best to revert to the released version of msprime while doing tsinfer dev.

hyanwong commented:
Just to follow up, tests are indeed failing on master if a bleeding edge msprime is installed:

```
test_multiple_recombinations (test_evaluation.TestInsertPerfectMutations) ... /usr/local/anaconda3/lib/python3.8/site-packages/msprime/intervals.py:187: UserWarning: num_loci is no longer supported and should not be used.
  warnings.warn("num_loci is no longer supported and should not be used.")
FAIL
```

jeromekelleher commented:
> Just to follow up, tests are indeed failing on master if a bleeding edge msprime is installed:

Yes - which is why I said to stick with the released version.

hyanwong commented:
Yes, sorry - messages crossed. Still, it will be annoying when tsinfer unit tests start failing because CI starts using msprime 1.0. So I was just sorting it out so that everything would carry on working.

hyanwong commented:
I'm just starting to use the msprime alpha for tsinfer testing. This squashes a few errors, but I'm still getting errors like 

```
>       assert len(tables.populations.metadata) == 0
E       AssertionError

tests/tsutil.py:48: AssertionError
```
 And indeed at

https://github.com/tskit-dev/tsinfer/blob/25312f75c3ecbe6de62a5380cceb4ac2df08b2fb/tests/tsutil.py#L42

it says "This will probably fail when msprime 1.0 comes along, but we can fix it then.".

I'm not clear how the permissive schemas etc conflict with msprime now, perhaps @benjeffery or @jeromekelleher can advise?

jeromekelleher commented:
> I'm not clear how the permissive schemas etc conflict with msprime now, perhaps @benjeffery or @jeromekelleher can advise?

Probably some details to be worked out, I wouldn't know without digging into it. Probably not worth messing with until msprime 1.0 is fully packaged though.

jeromekelleher commented:
Can you bring this up to date please @hyanwong?

hyanwong commented:
Ah, I remember the problem. Weirdly, even with the changes in here to remove mutation times, the new msprime is causing the following tests to fail:

```
FAILED tests/test_formats.py::TestAncestorData::test_ancestors_truncated_length - assert 431 == 213
FAILED tests/test_inference.py::TestMatchSamples::test_partial_samples - assert False
FAILED tests/test_inference.py::TestPathCompressionAncestorsPyEngine::test_simulation_with_error - assert 0 > 0
FAILED tests/test_inference.py::TestPathCompressionAncestorsCEngine::test_simulation_with_error - assert 0 > 0
```

I'll have a dig, but it's unclear to me what the problem is.

jeromekelleher commented on tests/test_evaluation.py:
```diff
@@ -397,15 +398,15 @@ def test_large_smc(self):
-        except TypeError:
-            # Fallback for older versions of msprime, which use num_loci
-            recomb_map = msprime.RecombinationMap.uniform_map(
+        except AttributeError:
```
No point in keeping this now, we should use either sim_ancestry or simulate

jeromekelleher commented on tests/tsutil.py:
```diff
@@ -28,6 +28,19 @@
+    tables = ts.dump_tables()
+    try:
+        tables.mutations.time = np.full_like(tables.mutations.time, tskit.UNKNOWN_TIME)
+    except AttributeError:
```
We're not interested in supporting older versions of msprime. We can add msprime>=1.0 to requirements.txt

jeromekelleher reviewed (commented):
Looks good, thanks. Can you update the requirements.txt files to require msprime > 1 and clear out the conditional stuff please?

hyanwong commented:
> Looks good, thanks. Can you update the requirements.txt files to require msprime > 1 and clear out the conditional stuff please?

Removed conditionals for old msprime and changed `requirements/development.txt` and the CI `requirements.txt` files where it was mentioned. Guess that's OK.

hyanwong commented:
CI now failing with 

```
ImportError: Legacy formats require h5py. Install via `pip install h5py` or `conda install h5py`
```

Any idea why that's suddenly appeared @jeromekelleher ? Do the requirements need changing?

jeromekelleher commented:
> ImportError: Legacy formats require h5py. Install via `pip install h5py` or `conda install h5py`

Not sure why we'd be reading in legacy files here @hyanwong, but it might simplest to put h5py into the various requirements.txt files for CI and development. (Either that or figure out where we're reading in hdf5 files and get rid of them - these are super old now and irrelevant for tsinfer.)

hyanwong commented:
> > ImportError: Legacy formats require h5py. Install via `pip install h5py` or `conda install h5py`
> 
> Not sure why we'd be reading in legacy files here @hyanwong, but it might simplest to put h5py into the various requirements.txt files for CI and development. (Either that or figure out where we're reading in hdf5 files and get rid of them - these are super old now and irrelevant for tsinfer.)

It's in `test_infer_bad_file` - I suspect we fall back to trying to load old formats.

jeromekelleher commented:
> It's in test_infer_bad_file - I suspect we fall back to trying to load old formats.

Just nuke the hdf5 files, they're irrelevant now

hyanwong commented:
There aren't any hdf5 files. We try to load a random file (e.g. a completely empty one), and fall back to using tskit if we can't recognise it. Tskit doesn't like the look of the file, so falls back to checking if it's hdf5 format. Bang.

jeromekelleher commented:
Ah - well, just add in the requirements then.

benjeffery commented on .github/workflows/wheels.yml:
```diff
@@ -145,7 +145,7 @@ jobs:
     strategy:
       matrix:
-        python: [3.6, 3.7, 3.8]
+        python: [3.7, 3.8]
```
3.9 was removed when we had issues with it - I think it can be reinstated now?

hyanwong replied:
Good point. Just added a commit to reinstate, seems fine.

benjeffery reviewed (approved):
LGTM - one question about wheel building.
